core/domain/change_domain.py

```diff
@@ -74,6 +74,14 @@ def validate_cmd(cmd_name, valid_cmd_attribute_specs, actual_cmd_attributes):
     if error_msg_list:
         raise utils.ValidationError((', ').join(error_msg_list))
 
+    deprecated_values = valid_cmd_attribute_specs.get('deprecated_values', {})
+    for attribute_name, attribute_values in deprecated_values.items():
+        actual_value = actual_cmd_attributes[attribute_name]
```

Comment:
what if the `attribute_name` is in `optional_attribute_names`? This will raise keyError?

Reply:
Yes and I think it should raise an error because even if the attribute_name is in optional_attribute_names, the value is still deprecated.

Reply:
Done.

Reply:
But as it's an optional attribute_name then why should we raise KeyError?

Reply:
I am sorry, I didn't understand it correctly before. Now corrected. Done.